PXB-193: When GTID_MODE=ON and slave uses MASTER_AUTO_POSITION=0, GTID is used in --slave-info

VasilyNemkov opened the pull request:
Using 'Auto_Position' and 'Using_Gtid' variables to determine GTID mode status for MySQL and MariaDB respectively.

param-build: http://jenkins.percona.com/job/percona-xtrabackup-2.4-param/219/
bug: https://jira.percona.com/browse/PXB-193

gl-sergei commented:
Fix is good. Just two things to consider
1. Provide fix for 2.3 as well
2. Cover the case when gtid=ON, but CHANGE MASTER TO would contain binlog coordinates in test suite

VasilyNemkov committed:
PXB-193: When GTID_MODE=ON and slave uses MASTER_AUTO_POSITION=0, GTID is used in --slave-info

Using 'Auto_Position' and 'Using_Gtid' variables to detect
GTID mode status for MySQL and MariaDB respectively.

VasilyNemkov committed:
Merge branch '2.3-PXB-193' into 2.4-PXB-193

VasilyNemkov commented:
Updated test.
2.3 was merged with https://github.com/percona/percona-xtrabackup/pull/380
[edit: was 381, the usual off-by-one-error]

gl-sergei reviewed (approved):
LGTM
